packages/yourloops/components/dialogs/pdf-print-options.tsx

```diff
@@ -0,0 +1,290 @@
+  const { open, onResult } = props;
+  const { t } = useTranslation("yourloops");
+  const theme = useTheme();
+  const matchLandscape = useMediaQuery(theme.breakpoints.up("sm"));
+  const fullScreen = useMediaQuery(theme.breakpoints.down("sm"));
+  const classes = printOptionsStyle();
+  const orientation: CalendarOrientation = matchLandscape ? "landscape" : "portrait";
+
+  const [customStartDate, setCustomStartDate] = React.useState<Dayjs|null>(null);
+  const { minDate, maxDate } = React.useMemo(() => {
+    let minDate = dayjs(props.minDate, { utc: true });
+    let maxDate = dayjs(props.maxDate, { utc: true });
+    if (customStartDate) {
+      const newMinDate = customStartDate.subtract(MAX_SELECTABLE_DAYS, "day");
+      const newMaxDate = customStartDate.add(MAX_SELECTABLE_DAYS, "day");
+      if (newMinDate.isAfter(minDate)) {
+        minDate = newMinDate;
+      }
+      if (newMaxDate.isBefore(maxDate)) {
+        maxDate = newMaxDate;
+      }
+    }
+    return { minDate, maxDate };
+  }, [props.minDate, props.maxDate, customStartDate]);
+
+  const [openState, setOpenState] = React.useState(false);
+  const [pdfOptions, setPDFOptions] = React.useState<PrintPDFOptions>(getDatesFromPreset(DEFAULT_PRESET, minDate, maxDate));
+
+  const { start, end, displayStart, displayEnd } = React.useMemo(() => {
+    const start = customStartDate ?? dayjs(pdfOptions.start, { utc: true });
+    const end = customStartDate ?? dayjs(pdfOptions.end, { utc: true });
+    const displayStart = start.format("ll");
+    const displayEnd = end.format("ll");
+    return { start, end, displayStart, displayEnd };
+  }, [pdfOptions, customStartDate]);
+
+  React.useEffect(() => {
+    // The openState is used to prevent to reset the selected dates, after a custom date selection
+    // Side effect of the useMemo for min/max dates.
+    // It's a workaround to mimic the prevProps param of React.Component.componentDidUpdate(prevProps)
+    if (open && !openState) {
+      setOpenState(true);
+      setPDFOptions(getDatesFromPreset(DEFAULT_PRESET, minDate, maxDate));
+    }
+    if (!open && openState) {
+      setOpenState(false);
+    }
+  }, [open, openState, minDate, maxDate]);
+
+  const handleClickPreset = (preset: Presets) => {
+    setPDFOptions(getDatesFromPreset(preset, minDate, maxDate));
+  };
+
+  const handleChangeCustomDate = (d: Dayjs) => {
+    if (customStartDate === null) {
+      setCustomStartDate(d);
+    } else {
+      const start = customStartDate.isBefore(d) ? customStartDate.format("YYYY-MM-DD") : d.format("YYYY-MM-DD");
+      const end = customStartDate.isBefore(d) ? d.format("YYYY-MM-DD") : customStartDate.format("YYYY-MM-DD");
+      setPDFOptions({ start, end });
+      setCustomStartDate(null);
+    }
+  };
+
+  const presetSelected = pdfOptions.preset;
+  const displayedDates = `${displayStart} → ${displayEnd}`;
+  return (
+    <Dialog
+      id="dialog-pdf-options"
+      fullScreen={fullScreen}
+      open={open}
```

Comment:
I don't see the use for this open, maybe I am missing something? 
Otherwise, it should be up to the parent component to decide wether or not to display this one

Reply:
Yes it's the case
The parent component choose the dialog state.